driver.go

```diff
@@ -29,8 +30,8 @@ type Driver struct {
 	cachedImage       *hcloud.Image
 	Type              string
 	cachedType        *hcloud.ServerType
-	Location          string
-	cachedLocation    *hcloud.Location
+	Locations         []string
```

Comment:
Personally I think the final location should still be in `Location`, as there are some downstream projects that depend on the 'public' data of the driver.

Keeping the location pool around in a separate public field seems like the best downward-compatible solution IMHO, that would still allow downstream tools to support this option in the future.